cli/cmd/cluster.go

```diff
@@ -107,38 +115,58 @@ func getLinkerdConfigMap() (*configPb.All, error) {
 	return global, nil
 }
 
-func buildMulticlusterSetupValues(opts *setupRemoteClusterOptions) (*multicluster.Values, error) {
+func buildMulticlusterInstallValues(opts *multiclusterInstallOptions) (*multicluster.Values, error) {
 
 	global, err := getLinkerdConfigMap()
 	if err != nil {
 		if kerrors.IsNotFound(err) {
-			return nil, errors.New("you need Linkerd to be installed in order to setup a remote cluster")
+			return nil, errors.New("you need Linkerd to be installed in order to install mutlicluster addons")
 		}
 		return nil, err
 	}
 
+	if !alphaNumDashDot.MatchString(opts.controlPlaneVersion) {
+		return nil, fmt.Errorf("%s is not a valid version", opts.controlPlaneVersion)
+	}
+
+	if opts.namespace == "" {
+		return nil, errors.New("you need to specify a namespace")
+	}
+
+	if opts.namespace == controlPlaneNamespace {
+		return nil, errors.New("you need to setup the multicluster addons in a namespace different than the Linkerd one")
+	}
+
+	if _, err := log.ParseLevel(opts.serviceMirrorLogLevel); err != nil {
+		return nil, fmt.Errorf("--log-level must be one of: panic, fatal, error, warn, info, debug")
```

Comment:
The flag in the error message wasn't updated (although might not need to according to my other suggestion).